src/render/Renderer.cpp

```diff
@@ -342,7 +342,7 @@ void CHyprRenderer::renderWindow(CWindow* pWindow, CMonitor* pMonitor, timespec*
 
             scaleBox(&windowBox, pMonitor->scale);
 
-            g_pHyprOpenGL->renderBorder(&windowBox, grad, rounding, a1);
+            g_pHyprOpenGL->renderBorder(&windowBox, grad, rounding, a1, pWindow->m_sSpecialRenderData.borderSize);
```

Comment:
if borderSize is not -1, pass it. Otherwise, pass the global config border size value. Render data contains additional render data, not persistent render data. It's meant for addons, not properties.

Reply:
Yeah, actually since this check is already done in the layout code, we might as well just use the `pWindow->m_sSpecialRenderData.borderSize` (which anyways defaults to -1). Since this function is called by other code though, we might also add the check here, I don't know if it makes sense though. What do you think?